ttnn/cpp/ttnn/operations/conv/conv2d/conv2d_pybind.cpp

```diff
@@ -57,9 +58,10 @@ void py_bind_conv2d(py::module& module) {
                 uint32_t groups,
                 std::optional<const ttnn::Tensor> bias_tensor,
                 std::optional<const Conv2dConfig> conv_config,
+                std::optional<const WormholeComputeKernelConfig> compute_config,
```

Comment:
you can't bind WormholeComputeKernelConfig here, you need to use a variant check how matmul is doing that
```suggestion
                std::optional<const WormholeComputeKernelConfig> compute_config,
```

Reply:
I see no advantage in having different types for WormhomeComputeKernelConfig and GrayskullComputeKernelConfig. 

GrayskullComputeKernelConfig is a subset of WormhomeComputeKernelConfig. The utility functions that are used to read individual params from the ComputeConfig already take in a device arch and handle unused params appropriately. 

Instead of raising an error when there is a mismatch between device arch and compute config type, it makes more sense to just return the appropriate default.

The current approach forces the model writer to check the device arch and pass the appropriate type of ComputeConfig. This would be better handled inside the C++ op implementation.

Reply:
All other ops work in this way, it's not ok for conv2d to do something else.
It's weird for users to have fixed wormhole confiig when they are for example working on Blackhole or Grayskull or some feature arch.

Reply:
If this is the correct approach, it would be best to change conv first, and change the other ops later.

Reply:
Pls keep conv inline with everything else.
If you do believe this is a better way to go, start a conversation about it on slack #tt-metal-developers or some other place.
If you reach an agreement that we should change this, make a plan and change it everywhere.

Reply:
I've reverted these changes. As discussed, we'll stick with DeviceComputeKernelConfig.